zio-schema/shared/src/main/scala/zio/schema/StandardType.scala

```diff
@@ -99,6 +108,112 @@ object StandardType {
     override def tag                                  = Tags.STRING
     override def compare(x: String, y: String): Int   = x.compareTo(y)
     override def defaultValue: Either[String, String] = Right("")
+
+    override def coerce[B](that: StandardType[B]): Option[String => Either[String, B]] =
+      that match {
+        case UnitType   => None
+        case StringType => Some(a => Right(a))
+        case BoolType   => Some(a => a.toBooleanOption.toRight(s"String $a could not be coerced into a Boolean"))
```

Comment:
Consider quoting the strings in all these cases, or you will get confusing error messages if the string has space characters.

e.g. if $a = "could not be", then the error message would be:

```
String could not be could not be coerced into a Boolean
```

But if you quote it:

```
String "could not be" could not be coerced into a Boolean
```